### PR TITLE
Enable express error handler

### PR DESCRIPTION
**Pull request by saulogt**
This PR is to enable the default express error handler on all cases if config param `enableExpressErrorHandler` is truthy

I'm solving the issue https://github.com/parse-community/parse-server/issues/4696 without changing the current default behavior.

### CODE REVIEW

**Comment by saulogt** [comment]
@flovilmart, That option `enableExpressErrorHandler` didn't exist. I introduced that just to alow (restore) the default express handler.
Test added.

**Comment by saulogt** [comment]
@flovilmart , is it still missing something?
I added the test as you requested.

File: spec/EnableExpressErrorHandler.spec.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+              expect(Parse.applicationId).toEqual("anOtherTestApp");
+              const app = express();
+              app.use('/parse', parseServer);
+
+              server = app.listen(12667);
+
+              app.use(function (err, req, res, next) {
+                next
+                lastError = err
+              })
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: Can you follow style and conventions and add the ; at the expected places please?

File: spec/EnableExpressErrorHandler.spec.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+const ParseServer = require("../src/index");
+const express = require('express');
+const rp = require('request-promise');
+
+describe('Enable express error handler', () => {
+  beforeEach((done) => {
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: Not sure why you need this, as you create a new server separately. Please remove if not required

File: spec/EnableExpressErrorHandler.spec.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }).then(() => {
+      done();
+    });
+  });
+
+  it('should call the default handler in case of error, like updating a non existing object', done => {
+    const serverUrl = "http://localhost:12667/parse"
+    const appId = "anOtherTestApp";
+    const masterKey = "anOtherTestMasterKey";
+    let server;
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: Seems written to, but never read.

File: spec/EnableExpressErrorHandler.spec.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          promise
+            .then(() => {
+              expect(Parse.applicationId).toEqual("anOtherTestApp");
+              const app = express();
+              app.use('/parse', parseServer);
+
+              server = app.listen(12667);
+
+              app.use(function (err, req, res, next) {
+                next
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: This line as no effect

File: spec/EnableExpressErrorHandler.spec.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                method: 'PUT',
+                uri: serverUrl + '/classes/AnyClass/nonExistingId',
+                headers: {
+                  'X-Parse-Application-Id': appId,
+                  'X-Parse-Master-Key': masterKey
+                },
+                body: { someField: "blablabla"},
+                json: true
+              })
+                .then(() => {
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: Can you keep the alignment consistent with the rest of the code? Better, use async / await

File: spec/EnableExpressErrorHandler.spec.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                  fail('Should throw error');
+                })
+                .catch(e => {
+                  expect(e).toBeDefined();
```
Comment (flovilmart) [inline spec/EnableExpressErrorHandler.spec.js]: Can we do some tests on the expected error here? As well as lastError?

File: spec/EnableExpressErrorHandler.spec.js (hunk 1 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: 👍

File: spec/EnableExpressErrorHandler.spec.js (hunk 2 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: Ok. Really not needed

File: spec/EnableExpressErrorHandler.spec.js (hunk 3 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: It's used to store the server to be closed later
```
.then(() => {
                  server.close(done);
                });
```

File: spec/EnableExpressErrorHandler.spec.js (hunk 4 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: This is just to silence the linter.
The function needs to have 4 arguments to be recognized as an error handler, but I don't want to call `next` again

File: spec/EnableExpressErrorHandler.spec.js (hunk 5 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: Keeping the alignment :)

File: spec/EnableExpressErrorHandler.spec.js (hunk 6 above)
Comment (saulogt) [reply spec/EnableExpressErrorHandler.spec.js]: Done

**Comment by saulogt** [comment]
Hey @flovilmart , Sorry about the delay.
I hope it's good now

**Comment by flovilmart** [comment]
Not sure why but the build fails on the CI.

**Comment by saulogt** [comment]
@flovilmart , CI fixed now. It was a recent change on `master` that got merged into my branch.
https://github.com/parse-community/parse-server/commit/ae1a8226d5613b6a5ba9564d53edced8b110cbf0

**Comment by flovilmart** [comment]
Perfect thanks!

**Pull request merged**